firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3683,7 +3682,8 @@ extension BrowserViewController: TabManagerDelegate {
 
         scrollController.tab = selectedTab
 
-        if let webView = selectedTab.webView {
+        if !selectedTab.isFxHomeTab,
```

Comment:
This is probably the only line of code actually needed.

Reply:
nice!! 

> The recent refactor to how redux actions execute surfaced a race condition where the webview was overwriting and removing the home page. This was previously happening in the opposite order so it wasn't causuing any obvious issues.

Do you mind expanding more on this and why the opposite order wasn't causing any obvious issues? The new bug seems like a consistent and reproducible, so curious on how the redux surface this race condition.

Reply:
Previously the redux action logic was dispatching async to the main thread more often, no it executes immediately more often. Before it was calling this block first, which sets the webview content and removes the homepage view controller from the embedded view and then the redux actions related to the homepage change were being handled, putting the new homepage in place. After the refactor the home page action is being handled immediately, putting the home page in place, and then it gets removed by this block of code updating the webview. 
Not sure if I described that well as it's a bit messy 😓

Reply:
this helps! thank you :)